consensus/types/round_state.go

```diff
@@ -131,7 +134,10 @@ func (rs *RoundState) RoundStateSimple() RoundStateSimple {
 // NewRoundEvent returns the RoundState with proposer information as an event.
 func (rs *RoundState) NewRoundEvent() types.EventDataNewRound {
 	addr := rs.Validators.GetProposer().Address
-	idx, _ := rs.Validators.GetByAddress(addr)
+	idx, _, err := rs.Validators.GetByAddress(addr)
+	if err != nil {
+		panic(err) // if proposer is not present in val set, panic
```

Comment:
@melekes not sure if its okay to panic here? Tried looking through the code to see where -1 gets checked (-1 was idx when val was not present before this pr) but can't find anywhere.

Reply:
It's okay to panic (at least for now). since we expect proposer to be in validator set